Upgrade to Hibernate ORM 5.1.2

langfr opened the pull request:
Let's give this a try first. :-)

ctomc committed:
WFLY-3639 default-web-module doesn't work for non default hosts & servers

- improve deployment message to include server on which it was deployed to
- add test to verify default-web-module works for all possible scenarios
- server addition doesn't require reload anymore

jmesnil committed:
[WFLY-5851] Add rebalance-connections to pooled-connection-factory resource

* update the XML schema and parser
* add transformer for legacy versions

JIRA: https://issues.jboss.org/browse/WFLY-5851

jmesnil committed:
[WFLY-3636] Add pool statistics for pooled-connection-factory

Add runtime resource statistics=pool to the pooled-connection-factory.
Statistics are enabled only if the pooled-connection-factory's
statistics-enabled attribute is true (it is false by default).

JIRA: https://issues.jboss.org/browse/WFLY-3636

pferraro committed:
WFLY-6602 Expose JGroups channels and channel factories as capabilities.
Replace ServiceName factories with equivalent capability/requirement ServiceName factories.

scottmarlow commented:
thanks Frank!  I'm looking forward to using  HHH-8706/HHH-10477 to bring us closer to solving the CDI/JPA cycle (WFLY-2387) which will be possible with ORM 5.1

scottmarlow commented:
Sigh, I wonder if the above failures will be addressed in ORM 5.1.1

Commit:
Fix CNFE because of missing com.fasterxml.classmate package.

langfr commented on jpa/hibernate5/src/main/java/org/jboss/as/jpa/hibernate5/HibernatePersistenceProviderAdaptor.java:
```diff
@@ -71,7 +71,7 @@ public void addProviderProperties(Map properties, PersistenceUnitMetadata pu) {
         putPropertyIfAbsent(pu, properties, AvailableSettings.KEYWORD_AUTO_QUOTING_ENABLED,"false");
         putPropertyIfAbsent(pu, properties, AvailableSettings.IMPLICIT_NAMING_STRATEGY, NAMING_STRATEGY_JPA_COMPLIANT_IMPL);
-        putPropertyIfAbsent(pu, properties, org.hibernate.ejb.AvailableSettings.SCANNER, HibernateArchiveScanner.class);
+        putPropertyIfAbsent(pu, properties, AvailableSettings.SCANNER, HibernateArchiveScanner.class);
```
Fixes 
INFO  [org.hibernate.orm.deprecation#<init>] HHH90000001: Found usage of deprecated setting for specifying Scanner [hibernate.ejb.resource_scanner]; use [hibernate.archive.scanner] instead

scottmarlow commented on pom.xml:
```diff
@@ -142,7 +142,7 @@
         <version.org.glassfish.javax.enterprise.concurrent>1.0</version.org.glassfish.javax.enterprise.concurrent>
         <version.org.glassfish.javax.json>1.0.3</version.org.glassfish.javax.json>
-        <version.org.hibernate>5.0.9.Final</version.org.hibernate>
+        <version.org.hibernate>5.1.0.Final</version.org.hibernate>
```
Could you try a temporary change to use "5.1.1-SNAPSHOT", instead of "5.1.0.Final"?  This will give us an idea of whether some of the test failures are fixed already.

scottmarlow commented:
Gail created HHH-10896 for the "attempt to set auto commit on a jta transaction" failure.

iweiss committed:
[WFLY-6784] Add possibility to enable websocket compression via management model

Issue: https://issues.jboss.org/browse/WFLY-6784

iweiss committed:
[WFLY-6784] Add possibility to enable websocket compression via management model

Issue: https://issues.jboss.org/browse/WFLY-6784

ehsavoie committed:
[WFCORE-429]: Incremental redeployment (single file update) over management API

Adding tests with exploded deployments and overlays.
Adding mixed-domain tests for exploded managed deployments.

bstansberry commented:
I put a fixme label on this as it references a snapshot. I assume deliberately. The label should get removed when the branch is updated.

Is this for 10.1 or 11? I assume 11.

bstansberry commented:
@scottmarlow I put 10.x on this so it will show up in filtered queries of open PRs and will be less likely to fall in a crack. But if it's going to be in 10.1 it needs to be working ASAP. If it for sure should be deferred to 11, please let me know and I'll change the label.

I'm not arguing one way or the other on getting it into 10.1; I have no personal opinion.

kabir commented:
@scottmarlow @bstansberry @Sanne 
I have removed the 10.x label, and added the 11.x and needs-kanban label. The reason for this is that it looks like it is related to https://issues.jboss.org/browse/EAP7-582 which I helped open earlier today. So it needs to go through the full Jirban workflow.

scottmarlow commented:
Looks like https://hibernate.atlassian.net/browse/HHH-10896 is still open, which I think is what the "fixme" of this request is regarding.

scottmarlow commented:
Regarding the agile/jirban/kanban workflow, I put the EAP7-582 issue into "Analysis Dev
The development team is analyzing the issue" mode.

scottmarlow commented:
Gail merged a change for https://hibernate.atlassian.net/browse/HHH-10896 to the Hibernate ORM 5.1 branch.  retest this please.

gbadner commented:
Some new commits were added for HHH-10896. Please re-test.

langfr commented:
Retest this please.

gbadner commented:
Retest this please.

gbadner commented:
Retest this please.

scottmarlow commented:
FYI, we need complete confirmation that ORM 5.1 is compatible with ORM 5.0.

scottmarlow commented:
Retest this please

gbadner commented:
I've created a new jira for this: https://hibernate.atlassian.net/browse/HHH-11024. The original jira (HHH-10896) was fixed, but a later commit for that same jira broke it again: 
https://github.com/hibernate/hibernate-orm/commit/076c7ea104f5bcd27a6958dd158a1742cf602cbf

scottmarlow commented:
Frank, could you try updating the org.jboss.as.test.integration.hibernate.SFSBHibernate2LcacheStats class to also specify:

`configuration.getProperties().put(AvailableSettings.TRANSACTION_COORDINATOR_STRATEGY, "jta"); `

Apparently the above ("hibernate.transaction.coordinator_class") setting is important to specify for native Hibernate ORM applications.

Commit:
Set TRANSACTION_COORDINATOR_STRATEGY to "jta".

langfr commented:
Retest this please

scottmarlow commented:
retest this please

scottmarlow commented:
I'm not sure why the testsuite isn't running.  Retest this please

scottmarlow commented:
Retest this please.

langfr commented:
CI seems to be down, does not respond. Site https://ci.wildfly.org/ is not reachable. :-(

scottmarlow commented:
Tomaz is investigating...

scottmarlow commented:
retest this please

scottmarlow commented:
retest this please

scottmarlow commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/SFSBHibernate2LcacheStats.java:
```diff
@@ -88,6 +88,7 @@ public void setupConfig() {
             configuration.setProperty(Environment.DATASOURCE, "java:jboss/datasources/ExampleDS");
             // set property to enable statistics
             configuration.setProperty("hibernate.generate_statistics", "true");
+            configuration.getProperties().put(AvailableSettings.TRANSACTION_COORDINATOR_STRATEGY, "jta");
```
This change helped but we also need the same line added to org.jboss.as.test.integration.hibernate.secondlevelcache.SFSB, after line 82.  This helped with my local testing.

Commit:
Set TRANSACTION_COORDINATOR_STRATEGY to "jta" (part II).

scottmarlow commented:
excellent to see this passing, thanks Frank! :)

rhusar committed:
WFLY-6965 WFLY-5239 Un-@Ignore intermittently failing XSiteSimpleTestCase#testPutRelayedToBackups

rhusar committed:
WFLY-6965 Fix and un-@Ignore RemoteEJB2ClientStatefulBeanFailoverDDTestCase previously failing due to JBPAPP-8726

rhusar committed:
WFLY-6965 Un-@Ignore previously failing ClusterPassivationTestCase due to JBPAPP-8774

Commit:
[WFLY-6980] Rename compensations impl package to internal

mkouba committed:
WFLY-6981 Upgrade Weld to 2.4.0.Final

- also remove the dependency on com.google.guava:guava
- also remove unused WeldAnnotationDiscovery

rhusar committed:
Revert "Add workaround for WFLY-5794 to infinispan tests"

This reverts commit 553957d3c48e054c5286a7d6f8d30e557918e7d9.

rhusar committed:
WFLY-7026 Remove test workaround for WFLY-5794

tomjenkinson committed:
JBTM-2748 Updated to use the new JTS JCA inflowed transaction recovery module

mmusgrov committed:
JBTM-2749 Upgrade jboss-transaction-spi to 7.5.0.Final

ochaloup committed:
[JBJCA-1173] Adding an integration test of JCA tracer functionality works when enabled

jmesnil committed:
Bump messaging-activemq version to 1.1.0

* update the messaging-activemq subsystem to 1.1.0 management version
* add 1.1 XML parser
* add resource transformers

jamezp committed:
[WFCORE-254] Removed the org.apache.commons.logging module as it will be provided by WildFly Core. Also removed the org.slf4j.jcl-over-slf4j module as well.

scottmarlow committed:
WFLY-7108 UNSYNCHRONIZED extended persistence context associated but not joined with JTA transaction and target component is persistence context of SYNCHRONIZED, IllegalStateException should be thrown

ctomc committed:
make build work on jdk9 b135+

scottmarlow committed:
WFLY-7075 add extension that allows mixed sync/unsync persistence context use or joined unsync persistence context

scottmarlow committed:
WFLY-7130 PersistenceProperty specified in @PersistenceContext should be passed to the created EntityManager

jamezp committed:
[WFLY-7132] Deprecate the org.slf4j.ext module as it's no longer required by any internal modules.

rachmatowicz committed:
[WFLY-7139] Handle "Component is stopped" exception on undeploy for EJB client.

stuartwdouglas committed:
WFLY-7144 Add option to reject HTTP/1.1 requests without a host header

psakar committed:
[WFLY-7038] License files issues

artemis-wildfly-integration license is LGPL 2.1

psakar committed:
[WFLY-7038] License files issues

jboss-ejb-api_3.2_spec license is GPL 2.0 with Classpath Exception

psakar committed:
[WFLY-7038] License files issues

jipijapa-hibernate5 license is ASL 2.0

jmartisk committed:
WFLY-7147 standardize access to node1's address in test suite and fix failing ClusteredJPA2LCTestCase

psakar committed:
[WFLY-7038] License files issues

gnu.getopt:java-getopt:1.0.13.redhat-5 license is GNU Library General Public License, Version 2

scottmarlow committed:
WFLY-7142 check for SynchronizationType.UNSYNCHRONIZED when processing persistence unit / context references from deployment descriptors

jmartisk committed:
WFLY-7160 test for all jca flush operations

ppalaga committed:
[WFLY-7183] 'name' attribute missing in XSD while required by web
subsystem parser

ctomc committed:
WFLY-6705 Namespaces / schema versions of module.xml are out of sync

soul2zimate committed:
[WFLY-7220] fix queue-size default value discrepancy

stuartwdouglas committed:
WFLY-7221 Batch registers control point before setting up batch environment

ctomc committed:
WFLY-7120 NPE in SessionManagerStatsHandle

ctomc committed:
#9221 org.hibernate:hibernate-search-analyzers is no longer used in build

darranl committed:
[WFLY-7230] Update the application-security-domain to use the 'name' attribute as used in the latest version of the schema.

stuartwdouglas committed:
WFLY-7221 Fix issue with earlier fix where control point would end up being null

pferraro committed:
Add convenience methods to Attribute.

pferraro committed:
Introduce ResourceDefinitionProvider to simplify enumerated ResourceDefinitions.

rhusar committed:
WFLY-7240 CacheRegistry is missing entries (e.g. client mappings) following a merge after a cluster split

tomjenkinson committed:
WFLY-7241 Upgrade Narayana to 5.3.5.Final

ctomc committed:
dont use sql logging in testsuite

kabir committed:
Revert "REVERTED - [WFLY-6784] Add possibility to enable websocket compression via management model"

bstansberry committed:
Upgrade WildFly Core to 3.0.0.Alpha9

hpehl committed:
WFLY-7239: Upgrade HAL to 2.9.0.Alpha3

kabir committed:
Revert "WFLY-3639 default-web-module doesn't work for non default hosts & servers"

jmesnil committed:
[WFLY-7256] Fix persistent configuration differences

* fix order of messaging and undertow XML attributes in the subsystem
  templates

JIRA: https://issues.jboss.org/browse/WFLY-7256

scottmarlow commented:
retest this please

scottmarlow commented:
@langfr looks like you need to sync up with a conflicting change made recently to WildFly master, could you please do something like "git pull upstream master" to initiate that, fix any conflicts and then push this branch again.  Thanks!

scottmarlow commented:
https://issues.jboss.org/browse/WFLY-6854
https://issues.jboss.org/browse/EAP7-582
https://issues.jboss.org/browse/JBEAP-5678

scottmarlow commented:
@bstansberry this change looks good to me except one of these commits is a merge, does that need to be changed to not be a "merge"?  I don't know the magic git commands to fix that, as when I tried to cherry-pick for EAP7-582, I get a "806d87febfb3e86e003e01ba08ca600d591daef7 is a merge but no -m option was given".  Should @langfr flatten his change or should I do that when I create a pr for EAP7-582?

bstansberry commented:
@scottmarlow Yes, the commit history should be cleaned up for this PR. For example I don't want us to have a commit moving to a snapshot in our history. Since fixing that will require a git rebase -i there's no need to keep the merge commit either.

scottmarlow commented:
@langfr can you please flatten your change?  Or you could create a new one based on https://patch-diff.githubusercontent.com/raw/wildfly/wildfly/pull/8984.patch, if you prefer.

langfr commented:
Replaced by https://github.com/wildfly/wildfly/pull/9257.
